arch/arm/src/nrf52/nrf52_spi.c

```diff
@@ -445,9 +456,13 @@ static int nrf52_spi_init(FAR struct nrf52_spidev_s *priv)
 
   pin   = GPIO_PIN_DECODE(priv->miso_pin);
   port  = GPIO_PORT_DECODE(priv->miso_pin);
+  regval = SPIM_PSELMISO_CONNECTED;  /* Disconnect */
+  if ((priv->miso_pin & GPIO_FUNC_MASK) != GPIO_DISABLED)
+    {
+      regval = (pin << SPIM_PSELMISO_PIN_SHIFT);
+      regval |= (port << SPIM_PSELMISO_PORT_SHIFT);
+    }
```

Comment:
```suggestion
#ifdef BOARD_SPI1_MISO_PIN
  pin   = GPIO_PIN_DECODE(BOARD_SPI1_MISO_PIN);
  port  = GPIO_PORT_DECODE(BOARD_SPI1_MISO_PIN);
  regval = (pin << SPIM_PSELMISO_PIN_SHIFT);
  regval |= (port << SPIM_PSELMISO_PORT_SHIFT);
#else 
  regval = SPIM_PSELMISO_CONNECTED;  /* Disconnect */
#endif
```

something like this?

Reply:
You cannot do that and keep the same control function for all the drivers.  Here you just made the pin configuration tied to `BOARD_SPI1_*_PIN` for SPI{0,1,2,3}.